File: 프로그래머스/lv1/42748. K번째수/K번째수.js
function solution(array, commands) {
  let answer = [];

  for (let i=0; i<commands.length; i++) {
    // let tmp = array.slice(commands[i][0]-1,commands[i][1]).sort()
    let tmp = array.slice(commands[i][0]-1,commands[i][1]).sort((a,b)=> a-b) // <- 이렇게 써야 통과!
    answer.push(tmp[commands[i][2]-1])
  }
  return answer;
}

// let tmp = array.slice(commands[i][0]-1,commands[i][1]).sort()
// 이거 tc 한개 통과 못함 ..? array의 각원소는 100까지임 ㅇㅇ. 100도 들어올 수 있음
// let a = [20, 100,10,15,51,15]
// a.sort()

// console.log(a) // [ 10, 100, 15, 15, 20, 51 ] -> 이렇게 됨!
// 와 진짜;; 명심하자 걍 sort는 항.상 원래 표현까지 전.부 써주자 숫자라고 방심해버렸네
// 문자열 sort할때는 뒤에꺼까지 제대로 써줬는데
// let tmp = array.slice(commands[i][0]-1,commands[i][1]).sort((a,b)=> a-b)

File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []

    for k in range(len(commands)):
        slice_arr = array[commands[k][0]-1:commands[k][1]]
        slice_arr.sort()
        
        answer.append(slice_arr[commands[k][2]-1])

    return answer


print(solution([1, 5, 2, 6, 3, 7, 4], [[2, 5, 3], [4, 4, 1], [1, 7, 3]]))

# # 레퍼런스1
# def solution(array, commands):
#     return list(map(lambda x:sorted(array[x[0]-1:x[1]])[x[2]-1], commands)
          
# # 레퍼런스2
# def solution(array, commands):
#     return [sorted(array[a-1:b])[c-1] for a,b,c in commands]